magento/magento2
# M2C-22090 Translate Wishlist and Compare label in template

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/372






### Description (*)

I checked this issue and right now these labels are created in backend when someone adds a product to Wishlist or Compare list. So if a product is added on English Store the label that is stored is 'item' or 'items'. Once a customer switches to German Store, the label is still stored as 'item' or 'items'. In order to update the label, the customer must add a new product (remove/update) to Wishlist or Compare list.

In my opinion keeping this label in backend is not the best idea. There might be a possibility to update customerData once the store has been changed but that would require backend resources, also we have translation options available both in knockout and phtml templates.

As the counter for Wishlist and Compare list is rendered with knockout, I tried to add span elements with `data-bind="i18n: '...'"` however I noticed that that didn't solve the issue while switching the store. I'm not quite sure but maybe the template is cached then. Using knockout's translate function doesn't work in phtml templates.

In the end I decided to check counter value with knockout and render different translation strings using Magento translation system in phtml templates.

### Fixed Issues (if relevant)

1. magento/magento2#24290: Wrong product label when changing store view

### Manual testing scenarios (*)

1. Create an account
2. Add a product to Wish List and Compare List
3. Switch store
4. Check if counter labels (e.g. '1 item', '4 items') are being translated properly

Please make sure to have two stores with different locale and translation packages installed.

![switcher](https://user-images.githubusercontent.com/34317830/68774512-81d90580-062d-11ea-8f52-3892443ae10b.gif)

### Questions or comments

I might need to update tests and I will do so.

### Contribution checklist (*)
